File: appserver-legacy/src/main/java/org/radarbase/appserver/dto/protocol/EmailNotificationProtocol.java
package org.radarbase.appserver.dto.protocol;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * @author yatharthranjan
 */
@Data
@AllArgsConstructor
@NoArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class EmailNotificationProtocol {
    
    @JsonProperty("enabled")
    private boolean enabled = false;

    @JsonProperty("title")
    private LanguageText title;

    @JsonProperty("text")
    private LanguageText body;
    
}

File: appserver-legacy/src/main/kotlin/org/radarbase/appserver/dto/protocol/EmailNotificationProtocol.kt
package org.radarbase.appserver.dto.protocol

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty

@JsonIgnoreProperties(ignoreUnknown = true)
data class EmailNotificationProtocol(
    @field:JsonProperty("enabled")
    val enabled: Boolean = false,

    @field:JsonProperty("title")
    var title: LanguageText? = null,

    @field:JsonProperty("text")
    var body: LanguageText? = null,
)
